### PR TITLE
Specify interaction bounds relative to the mod tile size.

### PR DESCRIPTION
**Pull request by pchote**
This PR changes the selection/interaction box sizes to be specified in WDist instead of pixels, making them independent of the mod artwork's resolution. This removes the need to specifically override selection box sizes when using the remastered assets.

### CODE REVIEW

**Comment by pchote** [comment]
Rebased and updated to use expression-bodied properties.

File: OpenRA.Mods.Common/Traits/Interactable.cs
Comment (teinarss) [inline OpenRA.Mods.Common/Traits/Interactable.cs:26]: Could it be an idea to have this as a explicit type, with the properties, width, height and optional offsets?

Comment (pchote) [reply OpenRA.Mods.Common/Traits/Interactable.cs:26]: I would be more keen on this if we had a way of dynamically registering types with FieldLoader. Right now we basically have to hardcode loadable types in `OpenRA.Game`*, which heavily discourages the use of any trait-specific types.

*We do have the `TypeConverter` stuff, but the plumbing overhead to implement this is larger than the types that would be using it.

**Comment by reaperrr** [comment]
Needs another rebase, LGTM after that.

**Comment by pchote** [comment]
Rebased.

File: OpenRA.Mods.Common/UpdateRules/Rules/20201213/ConvertBoundsToWDist.cs
Comment (teinarss) [inline OpenRA.Mods.Common/UpdateRules/Rules/20201213/ConvertBoundsToWDist.cs]: Not entirely true according to `var tileScale = grid.Type == MapGridType.RectangularIsometric ? 1448 : 1024;`?

**Comment by pchote** [comment]
Fixed and rebased.

**Review by teinarss — APPROVED** [review]

**Comment by teinarss** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/fc271cd9bb1713afa264680484c5cb10b462f62a...e37d05b78fdf25d97d43a49f508b26d1ac637a0c

**Pull request merged**
